src/SpaceStation.cpp

```diff
@@ -349,17 +336,17 @@ bool SpaceStation::GetDockingClearance(Ship *s, std::string &outMsg)
 
 	for (Uint32 i=0; i<m_shipDocking.size(); i++) {
 		// initial unoccupied check
-		if (m_shipDocking[i].ship != 0) continue;
+		if (m_shipDocking[i].GetShip() != nullptr ) continue;
 
 		// size-of-ship vs size-of-bay check
 		const SpaceStationType::SPort *const pPort = m_type->FindPortByBay(i);
 		if( !pPort ) continue;
 
 		if( pPort->minShipSize < bboxRad && bboxRad < pPort->maxShipSize ) {
 			shipDocking_t &sd = m_shipDocking[i];
-			sd.ship = s;
-			sd.stage = 1;
-			sd.stagePos = 0;
+			sd.StageClearanceGranted( s );
+			sd.EndRepos( false );
+			sd.EndRerot( false );
```

Comment:
Do we typically use spaces in the `(...)` like this? Looks odd to me, but I could be wrong. Worth checking out elsewhere in the codebase. Just do as others are doing.